src/OneinchCallee.sol

```diff
@@ -0,0 +1,158 @@
+pragma experimental ABIEncoderV2;
+
+import 'ds-test/test.sol';
+
+interface GemJoinLike {
+    function dec() external view returns (uint256);
+    function gem() external view returns (TokenLike);
+    function exit(address, uint256) external;
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (TokenLike);
+    function join(address, uint256) external;
+}
+
+interface TokenLike {
+    function approve(address, uint256) external;
+    function transfer(address, uint256) external;
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface CharterManagerLike {
+    function exit(address crop, address usr, uint256 val) external;
+}
+
+interface IAggregationExecutor {
+    function callBytes(address msgSender, bytes calldata data) external payable;
+}
+
+interface OneinchRouter {
+    struct SwapDescription {
+        TokenLike srcToken;
+        TokenLike dstToken;
+        address srcReceiver;
+        address dstReceiver;
+        uint256 amount;
+        uint256 minReturnAmount;
+        uint256 flags;
+    }
+
+    function swap(
+        IAggregationExecutor executor,
+        SwapDescription calldata desc,
+        bytes calldata permit,
+        bytes calldata data
+    ) external payable returns (uint256 returnAmount, uint256 spentAmount);
+}
+
+contract OneinchCallee {
+    DaiJoinLike public immutable daiJoin;
+    TokenLike public immutable dai;
+    uint256 public constant RAY = 10**27;
+
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, 'ds-math-add-overflow');
+    }
+
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, 'ds-math-sub-underflow');
+    }
+
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _add(x, _sub(y, 1)) / y;
+    }
+
+    constructor(address daiJoin_) public {
+        daiJoin = DaiJoinLike(daiJoin_);
+        TokenLike dai_ = DaiJoinLike(daiJoin_).dai();
+        dai = dai_;
+        dai_.approve(daiJoin_, uint256(-1));
+    }
+
+    function _fromWad(address gemJoin, uint256 wad) internal view returns (uint256 amt) {
+        amt = wad / 10**(_sub(18, GemJoinLike(gemJoin).dec()));
+    }
+
+    function clipperCall(
+        address sender, // Clipper caller, pays back the loan
+        uint256 owe, // Dai amount to pay back          [rad]
+        uint256 slice, // Gem amount received           [wad]
+        bytes calldata data // Extra data, see below
+    ) external {
+        (
+            address to, // address to send remaining DAI to
+            address gemJoin, // gemJoin adapter address
+            uint256 minProfit, // minimum profit in DAI to make [wad]
+            address charterManager, // pass address(0) if no manager
+            address router, // tx.to address received from the 1inch API
+        ) = abi.decode(data, (address, address, uint256, address, address, bytes));
```

Comment:
Well, 
- The data received from the 1inch is a calldata, so it includes function signature, therefore we can't directly pass it to `abi.decode`, we first need to remove first 4 bytes from it to be able to to pass to the decoder
- If we try to use the new variable decoded here, we first need to strip first 4 bytes, which is not possible due to the new variable being `memory` and `Index range access is only supported for dynamic calldata arrays` solidity error

Therefore, we end up working directly with calldata by removing all other values packed into the data (address, address, uint256, address, address) + 4 bytes. Using newly created variable to parse it later on will require to modify calldata returned from the 1inch, which we've tried to avoid in the first place. If this is important, I would then suggest to revert https://github.com/makerdao/exchange-callees/commit/11278ea359a465c1ecaf8da01e8795a14775c567

Reply:
As pointed out in the [other comment](https://github.com/makerdao/exchange-callees/pull/34#discussion_r1067219730): I had to find a compromise of removing first 4 bytes from the API response to address this request 🙂